tests/unit_tests/test_matchmaker_algorithm_team_matchmaker.py

```diff
@@ -253,16 +253,37 @@ def test_game_quality_time_bonus(s):
     team_a = CombinedSearch(*[s.pop(), s.pop(), s.pop()])
     c1 = CombinedSearch(*[s.pop(), s.pop()])
     team_b = CombinedSearch(*[c1, s.pop()])
-    quality_before = matchmaker.assign_game_quality((team_a, team_b)).quality
+    quality_before = matchmaker.assign_game_quality((team_a, team_b), 3).quality
 
     team_a.register_failed_matching_attempt()
     team_b.register_failed_matching_attempt()
-    quality_after = matchmaker.assign_game_quality((team_a, team_b)).quality
+    quality_after = matchmaker.assign_game_quality((team_a, team_b), 3).quality
 
-    num_newbies = sum(search.has_newbie() for search in team_a.get_original_searches())
-    num_newbies += sum(search.has_newbie() for search in team_b.get_original_searches())
+    num_newbies = team_a.num_newbies() + team_b.num_newbies()
 
-    assert quality_before + 6 * config.TIME_BONUS + num_newbies * config.NEWBIE_TIME_BONUS == pytest.approx(quality_after)
+    # player number / team size * time bonus
+    assert quality_before + 6 / 3 * config.TIME_BONUS + num_newbies / 3 * config.NEWBIE_TIME_BONUS \
+           == pytest.approx(quality_after)
+
+
+@given(st_searches_list(max_players=1, min_size=6, max_size=6))
+def test_game_quality_max_time_bonus(s):
+    matchmaker = TeamMatchMaker()
+    team_a = CombinedSearch(*[s.pop(), s.pop(), s.pop()])
+    c1 = CombinedSearch(*[s.pop(), s.pop()])
+    team_b = CombinedSearch(*[c1, s.pop()])
+    quality_before = matchmaker.assign_game_quality((team_a, team_b), 3).quality
+
+    for _ in range(30):
+        team_a.register_failed_matching_attempt()
+        team_b.register_failed_matching_attempt()
+    quality_after = matchmaker.assign_game_quality((team_a, team_b), 3).quality
+
+    num_newbies = team_a.num_newbies() + team_b.num_newbies()
+
+    # player number / team size * time bonus
+    assert quality_before + 6 / 3 * config.MAXIMUM_TIME_BONUS + num_newbies / 3 * config.MAXIMUM_NEWBIE_TIME_BONUS \
+           == pytest.approx(quality_after)
```

Comment:
You could parenthesize some of this so its more clear, plus I think that will let you split it across multiple lines more easily without using a backslash.